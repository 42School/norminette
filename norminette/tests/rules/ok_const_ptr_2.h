#ifndef OK_CONST_PTR_2_H
# define OK_CONST_PTR_2_H

int	execute_cmds(int const *const *cmds, char *envp[]);
int	execute_cmds(t_cmd const *const *cmds, char *envp[]);

int	func(int const *var);
int	func(const int *var);
int	func(int *const *const *const *const var);
int	func(int **const var);
int	func(int const *const **const var);

int	func(t_obj const *var);
int	func(const t_obj *var);
int	func(t_obj *const *const *const *const var);
int	func(t_obj **const var);
int	func(t_obj const *const **const var);

#endif
