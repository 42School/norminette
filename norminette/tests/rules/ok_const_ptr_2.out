[36mok_const_ptr_2.h[0m - [32mIsPreprocessorStatement[0m In "GlobalScope" from "None" line 1":
		<IFNDEF=#ifndef OK_CONST_PTR_2_H> <NEWLINE>
[36mok_const_ptr_2.h[0m - [32mIsPreprocessorStatement[0m In "GlobalScope" from "None" line 2":
		<DEFINE=# define OK_CONST_PTR_2_H> <NEWLINE>
[36mok_const_ptr_2.h[0m - [32mIsEmptyLine[0m In "GlobalScope" from "None" line 3":
		<NEWLINE>
[36mok_const_ptr_2.h[0m - [32mIsFuncPrototype[0m In "GlobalScope" from "None" line 4":
		<INT> <TAB> <IDENTIFIER=execute_cmds> <LPARENTHESIS> <INT> <SPACE> <CONST> <SPACE> <MULT> <CONST> <SPACE> <MULT> <IDENTIFIER=cmds> <COMMA> <SPACE> <CHAR> <SPACE> <MULT> <IDENTIFIER=envp> <LBRACKET> <RBRACKET> <RPARENTHESIS> <SEMI_COLON> <NEWLINE>
[36mok_const_ptr_2.h[0m - [32mIsFuncPrototype[0m In "GlobalScope" from "None" line 5":
		<INT> <TAB> <IDENTIFIER=execute_cmds> <LPARENTHESIS> <IDENTIFIER=t_cmd> <SPACE> <CONST> <SPACE> <MULT> <CONST> <SPACE> <MULT> <IDENTIFIER=cmds> <COMMA> <SPACE> <CHAR> <SPACE> <MULT> <IDENTIFIER=envp> <LBRACKET> <RBRACKET> <RPARENTHESIS> <SEMI_COLON> <NEWLINE>
[36mok_const_ptr_2.h[0m - [32mIsEmptyLine[0m In "GlobalScope" from "None" line 6":
		<NEWLINE>
[36mok_const_ptr_2.h[0m - [32mIsFuncPrototype[0m In "GlobalScope" from "None" line 7":
		<INT> <TAB> <IDENTIFIER=func> <LPARENTHESIS> <INT> <SPACE> <CONST> <SPACE> <MULT> <IDENTIFIER=var> <RPARENTHESIS> <SEMI_COLON> <NEWLINE>
[36mok_const_ptr_2.h[0m - [32mIsFuncPrototype[0m In "GlobalScope" from "None" line 8":
		<INT> <TAB> <IDENTIFIER=func> <LPARENTHESIS> <CONST> <SPACE> <INT> <SPACE> <MULT> <IDENTIFIER=var> <RPARENTHESIS> <SEMI_COLON> <NEWLINE>
[36mok_const_ptr_2.h[0m - [32mIsFuncPrototype[0m In "GlobalScope" from "None" line 9":
		<INT> <TAB> <IDENTIFIER=func> <LPARENTHESIS> <INT> <SPACE> <MULT> <CONST> <SPACE> <MULT> <CONST> <SPACE> <MULT> <CONST> <SPACE> <MULT> <CONST> <SPACE> <IDENTIFIER=var> <RPARENTHESIS> <SEMI_COLON> <NEWLINE>
[36mok_const_ptr_2.h[0m - [32mIsFuncPrototype[0m In "GlobalScope" from "None" line 10":
		<INT> <TAB> <IDENTIFIER=func> <LPARENTHESIS> <INT> <SPACE> <MULT> <MULT> <CONST> <SPACE> <IDENTIFIER=var> <RPARENTHESIS> <SEMI_COLON> <NEWLINE>
[36mok_const_ptr_2.h[0m - [32mIsFuncPrototype[0m In "GlobalScope" from "None" line 11":
		<INT> <TAB> <IDENTIFIER=func> <LPARENTHESIS> <INT> <SPACE> <CONST> <SPACE> <MULT> <CONST> <SPACE> <MULT> <MULT> <CONST> <SPACE> <IDENTIFIER=var> <RPARENTHESIS> <SEMI_COLON> <NEWLINE>
[36mok_const_ptr_2.h[0m - [32mIsEmptyLine[0m In "GlobalScope" from "None" line 12":
		<NEWLINE>
[36mok_const_ptr_2.h[0m - [32mIsFuncPrototype[0m In "GlobalScope" from "None" line 13":
		<INT> <TAB> <IDENTIFIER=func> <LPARENTHESIS> <IDENTIFIER=t_obj> <SPACE> <CONST> <SPACE> <MULT> <IDENTIFIER=var> <RPARENTHESIS> <SEMI_COLON> <NEWLINE>
[36mok_const_ptr_2.h[0m - [32mIsFuncPrototype[0m In "GlobalScope" from "None" line 14":
		<INT> <TAB> <IDENTIFIER=func> <LPARENTHESIS> <CONST> <SPACE> <IDENTIFIER=t_obj> <SPACE> <MULT> <IDENTIFIER=var> <RPARENTHESIS> <SEMI_COLON> <NEWLINE>
[36mok_const_ptr_2.h[0m - [32mIsFuncPrototype[0m In "GlobalScope" from "None" line 15":
		<INT> <TAB> <IDENTIFIER=func> <LPARENTHESIS> <IDENTIFIER=t_obj> <SPACE> <MULT> <CONST> <SPACE> <MULT> <CONST> <SPACE> <MULT> <CONST> <SPACE> <MULT> <CONST> <SPACE> <IDENTIFIER=var> <RPARENTHESIS> <SEMI_COLON> <NEWLINE>
[36mok_const_ptr_2.h[0m - [32mIsFuncPrototype[0m In "GlobalScope" from "None" line 16":
		<INT> <TAB> <IDENTIFIER=func> <LPARENTHESIS> <IDENTIFIER=t_obj> <SPACE> <MULT> <MULT> <CONST> <SPACE> <IDENTIFIER=var> <RPARENTHESIS> <SEMI_COLON> <NEWLINE>
[36mok_const_ptr_2.h[0m - [32mIsFuncPrototype[0m In "GlobalScope" from "None" line 17":
		<INT> <TAB> <IDENTIFIER=func> <LPARENTHESIS> <IDENTIFIER=t_obj> <SPACE> <CONST> <SPACE> <MULT> <CONST> <SPACE> <MULT> <MULT> <CONST> <SPACE> <IDENTIFIER=var> <RPARENTHESIS> <SEMI_COLON> <NEWLINE>
[36mok_const_ptr_2.h[0m - [32mIsEmptyLine[0m In "GlobalScope" from "None" line 18":
		<NEWLINE>
[36mok_const_ptr_2.h[0m - [32mIsPreprocessorStatement[0m In "GlobalScope" from "None" line 19":
		<ENDIF=#endif> <NEWLINE>
ok_const_ptr_2.h: OK!
